corehq/util/view_utils.py

```diff
@@ -69,7 +70,7 @@ def inner(request, *args, **kwargs):
         except BadRequest as e:
             return _get_json_exception_response(400, request, e)
         except Exception as e:
-            notify_exception(request, 'JSON exception response: {}'.format(e))
+            notify_exception(request, 'JSON exception response: {}'.format(smart_bytes(e)))
```

Comment:
👍 Moving downstreem to `notify_exception` could be nice, although I think you still need to use `smart_bytes` here as well because of the string formatting. Up to you if you want to go to that length.

For error handling like this I suppose `smart_bytes` is OK, considering the state of our world. However, it still feels worse to me than keeping track of the encoding of byte strings as we pass them around, or (easier) passing unicode strings. We still end up with effective nonsense if we pass around byte strings with unknown encoding, so `smart_bytes` is more of a band-aid than a magic solution to encodings.
```py
>>> from django.utils.encoding import smart_bytes
>>> print "Hey you, standing in the {}".format(smart_bytes(Exception(u"\xe9")))
Hey you, standing in the é
>>> print "Hey you, standing in the {}".format(smart_bytes(Exception("\xe9")))
Hey you, standing in the �
```